experimental/packages/otlp-exporter-base/src/platform/browser/OTLPExporterBrowserBase.ts

```diff
@@ -50,10 +51,15 @@ export abstract class OTLPExporterBrowserBase<
     this._contentType = contentType;
     this._useXHR =
       !!config.headers || typeof navigator.sendBeacon !== 'function';
+    if (typeof config.headers === 'function') {
+      this._getDynamicHeaders = config.headers;
```

Comment:
This does enable the bearer token use case, though it falls short for HMAC-signing since the body of the request is not available in the `getDynamicHeaders` function. Perhaps that would be a separate use case, although it is very similar.